Automattic/jetpack
# Experiment to export remote blocks

This is the Jetpack side of https://github.com/Automattic/wp-calypso/pull/35290

Export the Performance tab of Jetpack Settings as a remote block which can be displayed in Calypso.

#### Changes proposed in this Pull Request:

* Necessary infrastructure to build remote blocks
* APIs to describe remote blocks
* Sample implementation of `Performance` remote block

#### Testing instructions:
* Load up https://github.com/Automattic/wp-calypso/pull/35290 Calypso branch
* Tweak client/devdocs/design/remote.jsx in Calypso to replace goldsounds.ngrok.io with your own test site's URL
* Tweak the "authorization-credentials" meta tag in client/components/remote-block/index.jsx in Calypso so that it encodes the username/password for a valid user on your Jetpack site (see below for instructions)
* Head to http://calypso.localhost:3000/devdocs/remote
* After a delay, your component should display

##### Generating valid credentials

This PR doesn't include a final authentication mechanism. It includes a hand-rolled HTTP Basic Auth mechanism for authenticating API calls.

You can use PHP to generate a valid set of credentials like this:

```bash
$ wp shell
wp> base64_encode($username . ":" . $password);
(copy this output)
```
